### PR TITLE
New UI home [SERVER-2070]

### PR DESCRIPTION
**Pull request by Cookiezaurs**
*Home view (/core/home)
     ** Single view for 3 types(desktop, mobile, web)
     ** Passing widgets through mixins
     **  Downloading via /render
     ** User settings for which widgets to show in dashboard
     ** Changes in countly.template.js to use single view for mobile/desktop/web
*sessions  home widget (/clore/session-overview)
*events Home widget(/core/events)
*Sources and Keywords "widget" for home
*views home widget
*Technology home widget
*Home countries widget
*Crashes home widget

### CODE REVIEW

File: frontend/express/public/core/device-and-type/javascripts/countly.views.js [hunk 1]
```diff
@@ -227,4 +227,96 @@ countlyVue.container.registerTab("/analytics/technology", {
...
+        return {
+            dataBlocks: [],
+        };
+    },
+    beforeCreate: function() {
+        this.module = countlyDevicesAndTypes.getVuexModule();
+        CV.vuex.registerGlobally(this.module);
+    },
+    beforeDestroy: function() {
+        CV.vuex.unregister(this.module.name);
```
Comment (can-keklik) [inline frontend/express/public/core/device-and-type/javascripts/countly.views.js:246]: Could you please add `this.module = null` here?

File: frontend/express/public/core/device-and-type/javascripts/countly.views.js [hunk 2]
```diff
@@ -227,4 +227,96 @@ countlyVue.container.registerTab("/analytics/technology", {
...
+    }
+});
+
+countlyVue.container.registerData("/home/widgets", {
+    _id: "technology-dashboard-widget",
+    label: CV.i18n('sidebar.analytics.technology'),
+    description: CV.i18n('sidebar.analytics.technology-description'),
+    enabled: {"default": true}, //object. For each type set if by default enabled
+    available: {"default": true}, //object. default - for all app types. For other as specified.
+    order: 6, //sorted by ascending
```
Comment (can-keklik) [inline frontend/express/public/core/device-and-type/javascripts/countly.views.js:318]: We use "priority" by default, why "order" here? Do I miss something?

File: frontend/express/public/core/device-and-type/stylesheets/_main.scss [hunk 3]
```diff
@@ -8,4 +8,12 @@
 
 .version-graph-block:last-child {
 margin-bottom: 12px;
+}
+
+.technologyDashboardWidget>.bu-column:nth-child(odd)>.cly-vue-section{
```
Comment (can-keklik) [inline frontend/express/public/core/device-and-type/stylesheets/_main.scss:13]: Could you please replace camel case (technologyDashboardWidget) classes with kebab case (technology-dashboard-widget)? I mean, in order to maintain standards.

File: frontend/express/public/core/device-and-type/stylesheets/_main.scss [hunk 4]
```diff
@@ -8,4 +8,12 @@
...
+	padding-right:8px;
+}
+
+.technologyDashboardWidget>.bu-column:nth-child(even)>.cly-vue-section{
```
Comment (can-keklik) [inline frontend/express/public/core/device-and-type/stylesheets/_main.scss:17]: camel case issue here too

File: frontend/express/public/core/events/javascripts/countly.overview.views.js [hunk 5]
```diff
@@ -256,4 +256,50 @@
...
+            this.$store.dispatch('countlyEventsOverview/fetchTopEvents', 5).then(function() {
+                self.calculateAllData();
+            });
+        },
+        beforeCreate: function() {
+            this.module = countlyEventsOverview.getVuexModule();
+            CV.vuex.registerGlobally(this.module);
+        },
+        beforeDestroy: function() {
+            CV.vuex.unregister(this.module.name);
```
Comment (can-keklik) [inline frontend/express/public/core/events/javascripts/countly.overview.views.js:280]: Please add this.module = null here as well

File: frontend/express/public/core/events/javascripts/countly.overview.views.js [hunk 6]
```diff
@@ -256,4 +256,50 @@
...
+                this.items = data;
+            }
+        }
+    });
+    countlyVue.container.registerData("/home/widgets", {
+        _id: "events-dashboard-widget",
+        enabled: {"default": true}, //object. For each type set if by default enabled
+        available: {"default": true}, //object. default - for all app types. For other as specified.
+        placeBeforeDatePicker: true,
+        order: 2,
```
Comment (can-keklik) [inline frontend/express/public/core/events/javascripts/countly.overview.views.js:301]: Order/priority confusion here too

File: plugins/views/frontend/public/stylesheets/main.css [hunk 7]
```diff
@@ -6,3 +6,17 @@
 .routename-views-home .views-date-picker-container {
   margin-bottom: 32px;
 }
+
+.bu-is-6 .viewsDashboardWidget .cly-vue-metric-cards .cly-vue-metric-card {
```
Comment (can-keklik) [inline plugins/views/frontend/public/stylesheets/main.css:10]: A couple of things here:

-First I'm not sure if it is a good idea to limit the rule to bu-is-6. Any specific reasons for that?
-camelCase / kebab-case confusion here as well. 
-.cly-vue-metric-cards and .cly-vue-metric-card being overridden is a small violation but I guess you had no choice, right?

All these apply to following rules

File: plugins/views/frontend/public/javascripts/countly.views.js [hunk 8]
```diff
@@ -589,6 +660,21 @@
...
+        countlyVue.container.registerData("/home/widgets", {
+            _id: "views-dashboard-widget",
+            label: CV.i18n('views.title'),
+            description: CV.i18n('views.title-desc'),
+            enabled: {"default": true}, //object. For each type set if by default enabled
+            available: {"default": true}, //object. default - for all app types. For other as specified.
+            placeBeforeDatePicker: false,
+            linkTo: {"label": CV.i18n('views.go-to-views'), "href": "#/analytics/views"},
+            width: 6,
+            order: 4,
```
Comment (can-keklik) [inline plugins/views/frontend/public/javascripts/countly.views.js:674]: order/priority thing

File: plugins/views/frontend/public/javascripts/countly.views.js [hunk 9]
```diff
@@ -552,6 +552,77 @@
...
+            self.$store.dispatch('countlyViews/fetchTotals').then(function() {
+                self.dataBlocks = self.calculateAllData();
+            });
+        },
+        beforeCreate: function() {
+            this.module = countlyViews.getVuexModule();
+            CV.vuex.registerGlobally(this.module);
+        },
+        beforeDestroy: function() {
+            CV.vuex.unregister(this.module.name);
```
Comment (can-keklik) [inline plugins/views/frontend/public/javascripts/countly.views.js:573]: this.module = null here please :slightly_smiling_face:

File: plugins/sources/frontend/public/stylesheets/main.css [hunk 10]
```diff
@@ -1,3 +1,19 @@
 .source-table tr{cursor: pointer;}
 .source-20{width: 20%;}
-.source-40{width: 40%;}
\ No newline at end of file
+.source-40{width: 40%;}
+
+.bu-is-6 .keywordsDashboardWidget .cly-vue-metric-cards .cly-vue-metric-card {
```
Comment (can-keklik) [inline plugins/sources/frontend/public/stylesheets/main.css:5]: A couple of things here:

-First I'm not sure if it is a good idea to limit the rule to bu-is-6. Any specific reasons for that?
-camelCase / kebab-case confusion here as well.
-.cly-vue-metric-cards and .cly-vue-metric-card being overridden is a small violation but I guess you had no choice, right?

All these apply to following rules

File: plugins/crashes/frontend/public/javascripts/countly.views.js [hunk 11]
```diff
@@ -945,6 +945,81 @@
...
+            this.$store.dispatch("countlyCrashes/overview/refresh").then(function() {
+                self.calculateAllData();
+            });
+        },
+        beforeCreate: function() {
+            this.module = countlyCrashes.getVuexModule();
+            CV.vuex.registerGlobally(this.module);
+        },
+        beforeDestroy: function() {
+            CV.vuex.unregister(this.module.name);
```
Comment (can-keklik) [inline plugins/crashes/frontend/public/javascripts/countly.views.js:967]: this.module = null here please :slightly_smiling_face:

File: frontend/express/public/localization/dashboard/dashboard.properties [hunk 12]
```diff
@@ -350,13 +360,21 @@ sidebar.feedback = Feedback
 
 #dashboard
 dashboard.apply = Apply
-dashboard.avg-time-spent = AVG. SESSION DURATION
-dashboard.time-spent = TIME SPENT
+dashboard.home-desc = All Home things overview
```
Comment (can-keklik) [inline frontend/express/public/localization/dashboard/dashboard.properties:363]: "Home things" sounds like a filler text. Doesn't it? :slightly_smiling_face:

File: frontend/express/public/core/home/javascripts/countly.models.js [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+                return CV.$.ajax({
+                    type: "POST",
+                    url: countlyCommon.API_PARTS.users.w + "/updateHomeSettings",
+                    data: {
+                        app_id: countlyCommon.ACTIVE_APP_ID,
+                        homeSettings: JSON.stringify(data)
+                    },
+                    dataType: "json"
+                }, {disableAutoCatch: true}).then(function(/*json*/) {
+
```
Comment (can-keklik) [inline frontend/express/public/core/home/javascripts/countly.models.js:41]: So you fetch it but not interested in the response? :smile:

File: frontend/express/public/core/home/javascripts/countly.views.js [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+            countlyGlobal.member.homeSettings = countlyGlobal.member.homeSettings || {};
+            countlyGlobal.member.homeSettings[countlyCommon.ACTIVE_APP_ID] = userSettings;
+
+            this.$store.dispatch('countlyHomeView/updateHomeView', userSettings).then(function() {
+                CountlyHelpers.notify({type: "ok", title: jQuery.i18n.map["common.success"], message: jQuery.i18n.map["events.general.changes-saved"], sticky: false, clearAll: true});
+            });
+
+            this.calculatePlacedComponents();
+        },
+        Selected: function(command) {
```
Comment (can-keklik) [inline frontend/express/public/core/home/javascripts/countly.views.js:189]: "selected" would be better than "Selected". :slightly_smiling_face:

**Review by can-keklik — CHANGES_REQUESTED** [review]

File: frontend/express/public/core/device-and-type/javascripts/countly.views.js (hunk 2 above)
Comment (Cookiezaurs) [reply frontend/express/public/core/device-and-type/javascripts/countly.views.js:318]: Ignoring this one like you asked

File: frontend/express/public/core/events/javascripts/countly.overview.views.js (hunk 6 above)
Comment (Cookiezaurs) [reply frontend/express/public/core/events/javascripts/countly.overview.views.js:301]: Ignoring again?

**Pull request merged**